// Copyright (c) 2014 Baidu.com, Inc. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
syntax = "proto2";

import "google/protobuf/descriptor.proto";

package sofa.pbrpc;

extend google.protobuf.ServiceOptions {
    // Timeout in milli-seconds at service level. Default is 10 seconds.
    optional int64 service_timeout = 20000 [default = 10000];
}

enum CompressType {
    CompressTypeNone = 0;
    CompressTypeGzip = 1;
    CompressTypeZlib = 2;
    CompressTypeSnappy = 3;
    CompressTypeLZ4 = 4;
}

extend google.protobuf.MethodOptions {
    // Timeout in milli-seconds at method level. There is no default value because
    // method without method_timeout set will use service_timeout.
    optional int64 method_timeout = 20000;

    // Compress type. Default is no compression.
    optional CompressType request_compress_type = 20001 [default = CompressTypeNone];
    optional CompressType response_compress_type = 20002 [default = CompressTypeNone];
}
